src/lib/dnssd/BUILD.gn

```diff
@@ -14,14 +14,30 @@
 
 import("//build_overrides/chip.gni")
 import("//build_overrides/nlunit_test.gni")
+
+import("${chip_root}/build/chip/buildconfig_header.gni")
 import("${chip_root}/src/platform/device.gni")
 
 source_set("platform_header") {
   sources = [ "platform/Dnssd.h" ]
 }
 
+buildconfig_header("dnssd_buildconfig") {
+  header = "DnssdBuildConfig.h"
+  header_dir = "dnssd"
+
+  chip_dnssd_minimal = chip_mdns == "minimal"
+  chip_dnssd_platform = chip_mdns == "platform"
+
+  defines = [
+    "CHIP_DNSSD_MINIMAL=${chip_dnssd_minimal}",
```

Comment:
This seems to introduce a strong coupling between type of dnssd implementation and other code.

Could we instead have dns headers provide appropriate query methods? We tried to abstract away DNSSD implementations and provide a common set of headers. Having ifdefs seems to work against that abstraction.

Reply:
Yeah, I agree. I'll try to propose something else. Overall, I feel like we may need another injectable ServerReadyStrategy object to determine when to start outgoing communication, but let's see how much time I'll find for that.

Reply:
I got rid off those ifdefs and added `IsInitialized` method to the resolver. Please take another look.